### DISCUSSION TITLE
react-pdf utf8 font not working when on-fly-rendering, but works fine when render

### DISCUSSION THREAD

**Discussion by icepablo** [diegomura/react-pdf#1583] [Q&A] [answered]
My problem is i'm trying to save pdf file with 'on-fly-rendering' and utf8 font not working. But when displaying same pdf in render works fine.
Showing and saving pdf:
```
         <PDFViewer>
                <MyDocument data={data} />
              </PDFViewer>
              <PDFDownloadLink document={<MyDocument data={data}/>} fileName="lista_składników.pdf">
            {({ blob, url, loading, error }) => (loading ? 'Loading document...' : 'zapisz pdf')}
          </PDFDownloadLink>
```
Two different outputs:
![](https://i.stack.imgur.com/C1iEZ.png)
![](https://i.stack.imgur.com/j0uBD.png)

**Comment by michal-perlakowski** [comment, marked as answer]
https://github.com/diegomura/react-pdf/issues/588#issuecomment-491538579
